corehq/util/couch.py

```diff
@@ -33,3 +36,130 @@ def categorize_bulk_save_errors(error):
         result_map[error].append(result)
 
     return result_map
+
+
+class IterDB(object):
+    """
+    Bulk save docs in chunks.
+
+        with IterDB(db) as iter_db:
+            for doc in iter_docs(db, ids):
+                iter_db.save(doc)
+
+        iter_db.error_ids  # docs that errored
+        iter_db.saved_ids  # docs that saved correctly
+    """
+    def __init__(self, database, chunksize=100, throttle_secs=None):
+        self.db = database
+        self.chunksize = chunksize
+        self.throttle_secs = throttle_secs
+        self.saved_ids = []
+        self.deleted_ids = []
+        self.error_ids = []
+
+    def __enter__(self):
+        self.to_save = []
+        self.to_delete = []
+        return self
+
+    def commit(self, cmd, docs):
```

Comment:
this function seems like it should be made internal (prefixed with an `_`). 

I expected `commit()` to just be a shortcut for `commit_save()` + `commit_delete()`